lib/translator/status.cc

```diff
@@ -0,0 +1,243 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "status.h"
+
+#include "common.h"
+
+namespace translator {
+
+namespace {  // anonymous namespace for helper functions
+
+// Section 7.1
+// https://www.nvmexpress.org/wp-content/uploads/NVM-Express-SCSI-Translation-Reference-1_1-Gold.pdf
+void StatusToScsi(nvme_defs::GenericCommandStatusCode sc, ScsiStatus& result) {
```

Comment:
nit: prefer return values over mutable func parameters.

Reply:
these are just helper functions and not part of the interface. Nonetheless, would you still prefer I make the change?

Reply:
It's the preferred style whether or not it's public-facing, so long as it's practical to accomplish. So long as we're not returning a status code or similar here it'll make the code more readable.

Reply:
Okay, good to know! I'll make the change